tests/execution/graphql-conformance-015.md

```diff
@@ -0,0 +1,185 @@
+  searchComments(query: [[String!]!]! = [["today"]]): String! @expr(body: "video_{{.value.id}}_{{.args.query}}")
+}
+
+input VideoSize {
+  width: Int
+  height: Int
+  hdr: Boolean = true
+}
+```
+
+```yml @mock
+- request:
+    method: POST
+    url: http://upstream/graphql
+    textBody: '{ "query": "query { user(id: 4) { id name } }" }'
+  expectedHits: 9
+  response:
+    status: 200
+    body:
+      data:
+        user:
+          id: 4
+          name: User 4
+```
+
+```yml @test
+# Positve: no optional
+- method: POST
+  url: http://localhost:8080/graphql
+  body:
+    query: |
+      query {
+        user(id: 4) {
+          id
+          name
+          profilePic
+        }
+      }
+# Positve: different size
+- method: POST
+  url: http://localhost:8080/graphql
+  body:
+    query: |
+      query {
+        user(id: 4) {
+          id
+          name
+          profilePic(size: 200)
+        }
+      }
+# Positve: width only
+- method: POST
+  url: http://localhost:8080/graphql
+  body:
+    query: |
+      query {
+        user(id: 4) {
+          id
+          name
+          profilePic(width: 200)
+        }
+      }
+# Positve: width only, unset height
+- method: POST
+  url: http://localhost:8080/graphql
+  body:
+    query: |
+      query {
+        user(id: 4) {
+          id
+          name
+          profilePic(width: 200, height: null)
+        }
+      }
+# Positve: width and height
+- method: POST
+  url: http://localhost:8080/graphql
+  body:
+    query: |
+      query {
+        user(id: 4) {
+          id
+          name
+          profilePic(width: 200, height: 50)
+        }
+      }
+# Positve: video default
+- method: POST
+  url: http://localhost:8080/graphql
+  body:
+    query: |
+      query {
+        user(id: 4) {
+          id
+          name
+          featuredVideo
+        }
+      }
+# Positve: video overwrite
+- method: POST
+  url: http://localhost:8080/graphql
+  body:
+    query: |
+      query {
+        user(id: 4) {
+          id
+          name
+          featuredVideo(video: {width: 1920, height: 1080, hdr: true})
+        }
+      }
+# Positve: comments default
+- method: POST
+  url: http://localhost:8080/graphql
+  body:
+    query: |
+      query {
+        user(id: 4) {
+          id
+          name
+          searchComments
+        }
+      }
+# Positve: comments overwrite
+- method: POST
+  url: http://localhost:8080/graphql
+  body:
+    query: |
+      query {
+        user(id: 4) {
+          id
+          name
+          searchComments(query: [["test", "tost"], ["foo"], ["bar"], ["bizz", "buzz"]])
+        }
+      }
+
+# # Positve: defaults from input
+# Disabled because Tailcall does not uses default values from Input Object
+# - method: POST
+#   url: http://localhost:8080/graphql
+#   body:
+#     query: |
+#       query {
+#         user(id: 4) {
+#           id
+#           name
+#           featuredVideoPreview
```

Comment:
is this test right?

the field set args as required and we need to provide that arg (or that is validation missed case?) and we need to test that `hdr` is got defaulted to `true`

Reply:
changed it to `featuredVideoPreview(video: VideoSize! = {}): String!`. I set up another graphql implementation and verified that hdr is true. I disable the test for now.